experimental/LieAlgebras/src/AbstractLieAlgebra.jl

```diff
@@ -0,0 +1,155 @@
+          iszero,
+          struct_consts[i, j][k] + struct_consts[j, i][k] for i in 1:dimL, j in 1:dimL,
+          k in 1:dimL
+        ) "Not anti-symmetric."
+        @req all(
+          iszero,
+          sum(
+            struct_consts[i, j][k] * struct_consts[k, l][m] +
+            struct_consts[j, l][k] * struct_consts[k, i][m] +
+            struct_consts[l, i][k] * struct_consts[k, j][m] for k in 1:dimL
+          ) for i in 1:dimL, j in 1:dimL, l in 1:dimL, m in 1:dimL
+        ) "Jacobi identity does not hold."
+      end
+      new{C}(R, dimL, struct_consts, s)
+    end::AbstractLieAlgebra{C}
+  end
+end
+
+const AbstractLieAlgebraDict = CacheDictType{
+  Tuple{Ring,Matrix{SRow},Vector{Symbol}},AbstractLieAlgebra
+}()
+
+struct AbstractLieAlgebraElem{C<:RingElement} <: LieAlgebraElem{C}
+  parent::AbstractLieAlgebra{C}
+  mat::MatElem{C}
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+parent_type(::Type{AbstractLieAlgebraElem{C}}) where {C<:RingElement} =
+  AbstractLieAlgebra{C}
+
+elem_type(::Type{AbstractLieAlgebra{C}}) where {C<:RingElement} = AbstractLieAlgebraElem{C}
+
+parent(x::AbstractLieAlgebraElem{C}) where {C<:RingElement} = x.parent
+
+base_ring(L::AbstractLieAlgebra{C}) where {C<:RingElement} = L.R::parent_type(C)
+
+dim(L::AbstractLieAlgebra{C}) where {C<:RingElement} = L.dim
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+
+function Base.show(io::IO, V::AbstractLieAlgebra{C}) where {C<:RingElement}
+  print(io, "AbstractLieAlgebra over ")
+  print(IOContext(io, :compact => true), base_ring(V))
+end
+
+function symbols(L::AbstractLieAlgebra{C}) where {C<:RingElement}
+  return L.s
+end
+
+###############################################################################
+#
+#   Parent object call overload
+#
+###############################################################################
+
+# no special ones
+
+###############################################################################
+#
+#   Arithmetic operations
+#
+###############################################################################
+
+function bracket(
+  x::AbstractLieAlgebraElem{C}, y::AbstractLieAlgebraElem{C}
+) where {C<:RingElement}
+  check_parent(x, y)
+  L = parent(x)
+  mat = sum(
+    cxi * cyj * L.struct_consts[i, j] for (i, cxi) in enumerate(Generic._matrix(x)),
+    (j, cyj) in enumerate(Generic._matrix(y))
+  )
+  return L(mat)
+end
+
+###############################################################################
+#
+#   Constructor
+#
+###############################################################################
+
+function lie_algebra(
+  R::Ring,
+  struct_consts::Matrix{SRow{C}},
+  s::Vector{<:Union{AbstractString,Char,Symbol}}=[
```

Comment:
Could use the new `AbstractAlgebra.VarName` here which we introduced just for this purpose
```suggestion
  s::Vector{<:VarName}=[
```

Reply:
also in other places

Reply:
Changed in b2ec38dfec94a460f72a261ffb3771d72f662c78.